Sources/SwiftSyntax/SyntaxConvenienceMethods.swift

```diff
@@ -0,0 +1,81 @@
+//===- SyntaxConvenienceMethods.swift - Convenience funcs for syntax nodes ===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+public extension FloatLiteralExprSyntax {
+  var floatingValue: Double {
+    let floatingDigitsWithoutUnderscores = floatingDigits.text.filter {
+      $0 != "_"
+    }
+    return Double(floatingDigitsWithoutUnderscores)!
+  }
+}
+
+public extension IntegerLiteralExprSyntax {
+  var integerValue: Int {
+    let text = self.digits.text
+    let type = IntegerType(text: text)
+    let textWithoutPrefix: String
+    let digitsStartIndex = text.index(text.startIndex, offsetBy: type.prefixLength())
+    textWithoutPrefix = String(text.suffix(from: digitsStartIndex))
+
+    let textWithoutPrefixOrUnderscores = textWithoutPrefix.filter {
+      $0 != "_"
```

Comment:
Technically, `_` in each component is not valid. e.g.  `0x_12`, `0._12` and `12e_3`. But I think it's OK to accept them for now.
@akyrtzi What do you think?

Reply:
I think it's fine but adding a FIXME comment here would be good.

Reply:
@rintaro I observe that `_12` and `0x_2` don't parse as `IntegerLiteralExprSyntax` or `FloatLiteralExprSyntax`, respectively. My properties therefore shouldn't get those sorts of input. Should the properties nonetheless `fatalError()` if the `_` is the first character after the base prefix (or is the first character in the case of base 10)?

Reply:
As @akyrtzi said in another comment, we should prevent a case of creating syntax node with invalid literal text. If you do that, you don't need to check `_` in this `integerValue` or `floatingValue` accessor.

Also, when you check them, you need to check not only for the integral part, but you also need to check the fractional part and the exponent part. That's not so simple to implement in this PR. So I suggest you to add a `FIXME` comment as @akyrtzi said.

Reply:
I would say that for now it’s sufficient if we uphold the invariant that the contents will of a integer / float literal are always parsable from the `integerValue` resp. `floatingValue` properties, especially since the implementation is nice and short so far!

That should already be an improvement over the current behaviour and if we add a FIXME, we can come back to it and improve it later.